sources/typescript/src/adapters/kaltura-upload-connections-manager.ts

```diff
@@ -0,0 +1,71 @@
+export class KalturaUploadConnectionsManager {
+    private static _availableConnections: number;
+    private static _callbacks: (() => void)[];
+    private static _totalConnections: number = -1; // -1 as in not initialized
```

Comment:
the -1 is not needed anymore as the initialized state is managed in a dedicated flag